### ISSUE TITLE
Tree should support blur events

### ISSUE THREAD

**Issue by roieu** [devtools-html/debugger.html#2053]
Tree of elements, like Sources tree in the debugger, should be able to support a blur event to ensure we keep a selected node highlighted but ignoring any arrow navigation (mute onKeyDown/Press/Up events).
This relates to another issue, to support arrow navigation in the scope panel, which is currently impossible because Sources tree takes over any arrow keys navigation.

**Comment by jasonLaster** [comment]
Hi @roieu. I believe what you're saying is that when the source tree is blurred, it should be clear which source is selected. Is that correct?

**Comment by roieu** [comment]
@jasonLaster that is correct.
It also means that when the tree is not in focus it should stop respond to keyboard events.
Then we'll be able to have arrow keys navigation in Call Stack frame, which now no matter which
frame is in focus, the keyboard events are intercepted by the source tree control.
I am playing around with the tree in devtools-core to handle blur events (I am new to this project so my pace is slow)

**Comment by jasonLaster** [comment]
Great. Checkout how we do focus events in the call stack and blur in the
watch expression input

**Comment by jasonLaster** [comment]
@roieu any updates?

**Comment by roieu** [comment]
@jasonLaster Haven't got a chance to dive deep into it last week.
Hope to share something later this week.

**Comment by jasonLaster** [comment]
sounds good.

By the way, just noticed that the result list ui doesn't do much with blur events either...

**Comment by roieu** [comment]
@jasonLaster I have added blur support to the tree component and also added blur item style to the SourceTree.
Creating a pull request....

**Comment by jasonLaster** [comment]
Hey @roieu, how's it going?

It seems like this work might be blocked on a notion of "blurred". Feel free to hop in slack and chat about it.

**Comment by jasonLaster** [comment]
Some notes from the PR

adding blurred makes me concerned that we could be making things more difficult for ourselves in the future:  

* blurred could be a state for many elements
* blurred/focused might not be weird in some cases - an element is not focused, but it is blurred

I would like to better understand how atom does it. I started looking at their HTML:
One thing I do like is that it uses `::before` for the background which is nice

![](http://recordit.co/LzeyoWtBkG.gif)


also, here's the code: https://github.com/atom/tree-view

**Comment by jasonLaster** [comment]
closing this PR as we discussed a better way last week:

We should dull the selected item when the tree is not focused!
